commonmark/src/main/java/org/commonmark/parser/Parser.java

```diff
@@ -48,6 +50,16 @@ public Node parse(String input) {
         }
         return document;
     }
+    
+    public Node parse(Reader input) throws IOException {
```

Comment:
I'm now thinking maybe we shouldn't use overloading here. So maybe rename it to `parseReader` instead, so that it's clearly separate.